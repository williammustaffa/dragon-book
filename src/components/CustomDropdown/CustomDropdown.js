import React from "react";
import PropTypes from "prop-types";
import { Dropdown } from "semantic-ui-react";
import classNames from "classnames";

function CustomDropdown({ name, label, defaultValue, onChange, error, options }) {
  function handleChange(event, { value }) {
    if (typeof onChange === "function") {
      onChange(event, { name, value });
    }
  }

  return (
    <div className={classNames("custom-dropdown", "field", { error })}>
      {label && <label>{label}</label>}
      <Dropdown
        fluid
        selection
        name="type"
        defaultValue={defaultValue}
        onChange={handleChange}
        options={options}
      />
    </div>
  );
}

CustomDropdown.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  defaultValue: PropTypes.any,
  onChange: PropTypes.func,
  error: PropTypes.any,
  options: PropTypes.array.isRequired,
};

export default CustomDropdown;