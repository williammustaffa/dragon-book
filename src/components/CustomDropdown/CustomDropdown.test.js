import React from "react";
import CustomDropdown from "./CustomDropdown";
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

test("Test CustomDropdown component", () => {
  const { asFragment } = render(
    <CustomDropdown
      options={[{ key: "1", value: "1", test: "option" }]}
      name="test-dropdown"
    />
  );

  // Run tests
  expect(asFragment()).toMatchSnapshot();
});