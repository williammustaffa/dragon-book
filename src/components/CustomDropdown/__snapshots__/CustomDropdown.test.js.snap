// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CustomDropdown component 1`] = `
<DocumentFragment>
  <div
    class="custom-dropdown field"
  >
    <div
      aria-expanded="false"
      class="ui fluid selection dropdown"
      name="type"
      role="listbox"
      tabindex="0"
    >
      <div
        aria-atomic="true"
        aria-live="polite"
        class="text"
        role="alert"
      />
      <i
        aria-hidden="true"
        class="dropdown icon"
      />
      <div
        class="menu transition"
      >
        <div
          aria-checked="false"
          aria-selected="true"
          class="selected item"
          role="option"
          style="pointer-events: all;"
          test="option"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
